2014_TPDS_basic_decomposition_based_algorithm/src/handle_segment.rs

```diff
@@ -50,13 +57,55 @@ pub fn create_segments(dag: &mut Graph<NodeData, i32>) -> Vec<Segment> {
                 && segment.end_range <= node.params["earliest_finish_time"]
             {
                 segment.nodes.push(node.clone());
+                segment.parallel_degree += 1;
             }
         }
     }
 
     segments
 }
 
+fn classify_segment(volume: f32, period: f32, crit_path_len: f32, segment: &mut Segment) {
+    if segment.nodes.is_empty() {
+        unreachable!("Segment is empty")
+    }
+
+    segment.classification =
+        if segment.parallel_degree as f32 > volume / ((2.0 * period) - crit_path_len) {
+            Some(SegmentClassification::Heavy)
+        } else {
+            Some(SegmentClassification::Light)
+        };
+}
+
+#[allow(dead_code)] //TODO: remove
+fn classify_segments(
+    volume: f32,
+    period: f32,
+    crit_path_len: f32,
+    segments: &mut [Segment],
+) -> DAGClassification {
+    for segment in segments.iter_mut() {
+        classify_segment(volume, period, crit_path_len, segment);
+    }
+
+    let (mut heavy_count, mut light_count) = (0, 0);
+    for segment in segments {
+        match segment.classification {
+            Some(SegmentClassification::Heavy) => heavy_count += 1,
+            Some(SegmentClassification::Light) => light_count += 1,
+            None => unreachable!("Segment classification error"),
```

Comment:
```suggestion
            _ => unreachable!("Segment classification error"),
```

Reply:
Is the reason for `_` for scalability?
Changed.